src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -308,17 +308,23 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                                        var_types             retType,
                                        unsigned              simdSize)
 {
```

Comment:
Is the intrinsic *"special"* because it doesn't have a base type or because it has no arguments?
What prevents us from making it table-driven?

Reply:
because it doesn't have a base type and with there only being 3-4 instructions that don't, it's not been worth the more complex logic to make it table driven yet.

Reply:
I don't know how many intrinsics with no operands and no return types we are going to add in the future, so perhaps, it might make sense to extend our general logic to allow such intrinsics to be handled without making each of them *"special"*? @tannergooding What do you think?

Reply:
I expect there will be next to none. This is largely limited to fences + pause/yield. Everything else takes a parameter or returns "something" so I don't think its worth the additional churn in this PR.

Additionally, unlike many other cases they largely don't follow a standard pattern (that is, there is no relation between say a `mfence` and a `pause` instruction) and so often need some special handling regardless.